src/http_server.cc

```diff
@@ -3381,6 +3417,120 @@ HTTPAPIServer::HandleGenerate(
   request_release_payload.release();
 }
 
+#ifdef TRITON_ENABLE_METRICS
+std::vector<HTTPAPIServer::PromMetric>
+HTTPAPIServer::MetricFamilyExtractor(
+    const std::string& input, const std::string& metricFamily)
+{
+  std::vector<PromMetric> metrics;
+  // Construct the regex pattern using the provided metricFamily
+  std::string patternStr = metricFamily + R"((?:{(.*?)})?\s+(\d+(?:\.\d+)?))";
```

Comment:
Explain how the regex works. Given an example input and output

Reply:
Broke this regex down into into named variables and explained in comments with example.